src/azul/drs.py

```diff
@@ -74,10 +87,164 @@ class Access:
     headers: Optional[Mapping[str, str]] = None
 
 
+@attr.s()
+class DRSURI(metaclass=ABCMeta):
+
+    @property
+    @abstractmethod
+    def scheme(self) -> str:
+        raise NotImplementedError
```

Comment:
This is only used to facilitate the assert in the post-init hook so it should be protected, especially since having it public misleadingly suggests that it may return different values. But even making it protected would be overkill. I think we should just push down the scheme assertion, using whatever mechanism is appropriate there, and eliminate the attrs decorator from this class.